### PR TITLE
Add support for device_id 

### PR DESCRIPTION
**Pull request by vijaykramesh**
In addition to or instead of user_id

See https://github.com/toothrot/amplitude-api/issues/2

### CODE REVIEW

File: lib/amplitude_api/event.rb [hunk 1]
```diff
@@ -81,11 +86,17 @@ def to_hash
       serialized_event[:user_id] = user_id
       serialized_event[:event_properties] = event_properties
       serialized_event[:user_properties] = user_properties
+      serialized_event = add_optional_properties(serialized_event)
```
Comment (mdimas) [inline lib/amplitude_api/event.rb]: Curious why this wasn't adding optional properties before?

Comment (toothrot) [inline lib/amplitude_api/event.rb]: Same question!

**Comment by toothrot** [comment]
Awesome. Love the test coverage. I'll test this out and will work on getting it merged this weekend.

File: lib/amplitude_api/event.rb (hunk 1 above)
Comment (vijaykramesh) [inline lib/amplitude_api/event.rb]: rubocop complained when I added one more `serialized_event[:distinct_id] if distinct_id` to the method. so it seemed to make the most sense to pull out all the optional properties into their own method, vs telling rubocop to ignore that method.

**Comment by vijaykramesh** [comment]
http://imgur.com/a/3D31q has some verification screenshots from Amplitude, using the following code in an irb console on my branch to generate the events:

```ruby
event_only_device = AmplitudeAPI::Event.new({
  device_id: 'lorem',
  event_type: "vkr_test_with_device",
  time: Time.now,
  event_properties: {
    cause: "button",
    arbitrary: "properties"
  }
})
event_only_user = AmplitudeAPI::Event.new({
  user_id: '12345',
  event_type: "vkr_test_with_device",
  time: Time.now,
  event_properties: {
    cause: "button",
    arbitrary: "properties"
  }
})

event_user_and_device = AmplitudeAPI::Event.new({
  device_id: 'ipsum',
  user_id: '67890',
  event_type: "vkr_test_with_device",
  time: Time.now,
  event_properties: {
    cause: "button",
    arbitrary: "properties"
  }
})

AmplitudeAPI.track(event_only_device)
AmplitudeAPI.track(event_only_user)
AmplitudeAPI.track(event_user_and_device)
```

File: lib/amplitude_api/event.rb (hunk 1 above)
Comment (mdimas) [inline lib/amplitude_api/event.rb]: Ah got it, I was reading the diff incorrectly. `add_optional_properties` didn't use to exist! 👍

**Comment by toothrot** [comment]
I love everything about this pull request, especially the manual and automated testing.

**Pull request merged**